test/Streamly/Test/Data/Array/Foreign.hs

```diff
@@ -229,6 +248,21 @@ testAsPtrUnsafeMA = do
         rest <- getList (i + 1) (p `plusPtr` sizeOfInt)
         return $ val : rest
 
+reallocMA :: Property
+reallocMA =
+    forAll (vectorOf len (arbitrary :: Gen Char)) $ \vec ->
+      forAll (chooseInt (bSize - 100, bSize + 100)) $ \newBLen -> do
+          arr <- MA.fromList vec
+          arr1 <- MA.realloc newBLen arr
+          lst <- MA.toList arr
+          lst1 <- MA.toList arr1
+          lst `shouldBe` lst1
+
+    where
+
+    len = 10000
```

Comment:
Shouldn't this be length of `vec`? If the vector is larger than 10000 then it will get truncated after reallocation. If we want to test truncation as well then we should truncate the original list to the max length of the array before comparing.